Sources/NIO/BaseSocketChannel.swift

```diff
@@ -837,8 +849,19 @@ class BaseSocketChannel<SocketType: BaseSocketProtocol>: SelectableChannel, Chan
             self.pendingConnect = nil
             connectPromise.fail(error)
         }
+        
+        let pipelineError: Error?
+        switch error as? ChannelError {
+        case .some(ChannelError.inputClosed),
```

Comment:
this is obviously ugly, we have a similar issue elsewhere that we have "errors" that aren't errors.